modules/packages/ZMQ.chpl

```diff
@@ -0,0 +1,412 @@
+/*
+ * Copyright 2004-2016 Cray Inc.
+ * Other additional copyright holders may be indicated within.
+ *
+ * The entirety of this work is licensed under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ *
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// ZMQ.version
+// libzmq version 4.0.*
+
+/*
+  Module ZMQ: Chapel bindings for the ZeroMQ Messaging Library
+*/
+module ZMQ {
+
+  require "zmq.h", "-lzmq";
+
+  use Reflection;
+
+  /*
+    External ZMQ interface
+     - Currently, the special nature of Chapel strings means that we have to
+       choose between using pointer types (which can clear up C compilation
+       warnings) and strings.  This should go away with the new strings.
+     - Ideally, these would be "private" extern procs, which could only be
+       directly accessible inside module ZMQ.
```

Comment:
Yes, *two* out-of-date comments! We have new strings and private externs...and I use them! It seems some comment cleanup is in order.